spinoso-string/src/impls.rs

```diff
@@ -209,42 +221,42 @@ impl Deref for String {
 
     #[inline]
     fn deref(&self) -> &[u8] {
-        &*self.buf
+        &*self.inner.as_slice()
     }
 }
 
 impl DerefMut for String {
     #[inline]
     fn deref_mut(&mut self) -> &mut [u8] {
-        &mut *self.buf
+        &mut *self.inner.as_mut_slice()
     }
 }
 
 impl Borrow<[u8]> for String {
     #[inline]
     fn borrow(&self) -> &[u8] {
-        self.buf.as_slice()
+        self.inner.as_slice()
     }
 }
 
 impl BorrowMut<[u8]> for String {
     #[inline]
     fn borrow_mut(&mut self) -> &mut [u8] {
-        self.buf.as_mut_slice()
+        self.inner.as_mut_slice()
     }
 }
 
 impl Borrow<Vec<u8>> for String {
     #[inline]
     fn borrow(&self) -> &Vec<u8> {
-        &self.buf
+        self.inner.as_vec()
```

Comment:
This will help us get rid of `as_vec`.

```suggestion
        self.inner.borrow()
```